src/core/shape_inference/include/nms_shape_inference.hpp

```diff
@@ -68,18 +69,21 @@ void shape_infer(const NonMaxSuppression* op,
     ov::PartialShape out_shape = {Dimension::dynamic(), 3};
     if (boxes_ps.rank().is_static() && scores_ps.rank().is_static()) {
         const auto num_boxes_boxes = boxes_ps[1];
-        if (num_boxes_boxes.is_static() && scores_ps[0].is_static() && scores_ps[1].is_static()) {
-            const auto num_boxes = num_boxes_boxes.get_length();
-            const auto num_classes = scores_ps[1].get_length();
+        bool gpu_wa =
```

Comment:
do we plan to remove this WA in future and have alignment between CPU / GPU?

Reply:
CPU has custom code which delivers exact output shape from the execution kernel and don't need no shape inference. GPU on the other hand has two ways of NMS execution -- static way via padding with zero and dynamic. for static case -- they need some piece of code to generate static output shape which differs from original definition of NMS. In my opinion they need to get rid of static execution of NMS at all. But if it isn't possible -- I created a ticket for GPU to adapt their internal code -- 107654
So either they resolve the ticket whatever way possible, or they get rid of NMSIE usage.